### PR TITLE
Chore: RuboCop Style/MultipleComparison

### PR DESCRIPTION
**Pull request by Drenmi**
## Description

Part of [RuboCop Quest](https://github.com/lostisland/faraday/issues/854).

This fixes the single instance of the offense for `MultipleComparison`. It turned out that the code wasn't needed at all, since `Faraday::Utils.escape` already correctly stringifies booleans.

## Additional Notes

I also added a unit test that seemed to be missing for this code.

### CODE REVIEW

File: lib/faraday/encoders/flat_params_encoder.rb [hunk 1]
```diff
@@ -29,7 +29,6 @@ def self.encode(params)
       buffer = +''
       params.each do |key, value|
         encoded_key = escape(key)
-        value = value.to_s if value == true || value == false
```
Comment (Drenmi) [inline lib/faraday/encoders/flat_params_encoder.rb]: This is handled by `Faraday::Utils.escape` already:

```ruby
Faraday::Utils.escape(true)
#=> "true"
```

Comment (olleolleolle) [reply lib/faraday/encoders/flat_params_encoder.rb]: I got curious about the test for this, where a `value` is a Boolean:

`.encode([{ 'a' => true, 'b' => false}])`

Perhaps we want that test, too?

Comment (iMacTia) [reply lib/faraday/encoders/flat_params_encoder.rb]: Agree with @olleolleolle, we need a test that tests `encode` method with boolean values, if there isn't one already.

Comment (Drenmi) [reply lib/faraday/encoders/flat_params_encoder.rb]: I might have been a bit too quick. Will have a look at it tonight. 🙂

Comment (iMacTia) [reply lib/faraday/encoders/flat_params_encoder.rb]: After adding some more tests, it looks like this line really is redundant.

**Pull request merged**